### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by velppa**
Changelog:
- Handle all errors
- Pass errors to `main` so app can only exit from there
- Wait for spawned goroutines to complete

### CODE REVIEW

File: underblog.go [hunk 1]
```diff
@@ -67,128 +78,138 @@ func main() {
...
+	return t.Execute(f, post)
 }
 
-func FnameWithoutExtension(fn string) string {
+func fnameWithoutExtension(fn string) string {
 	return strings.TrimSuffix(fn, path.Ext(fn))
 }
 
-func VerifyFilenameBaseFormat(f string) {
-	errorDescription := "I can't parse this filename. Make sure its name is formatted as: DD-MM-YYY-slug.md"
```
Comment (freetonik) [inline underblog.go:145]: While your version is indeed shorted and faster to read, first person was a deliberate choice for humane effect. I'd love it to stay that way.

File: underblog.go [hunk 2]
```diff
@@ -67,128 +78,138 @@ func main() {
...
+func verifyFilenameBaseFormat(f string) error {
+	filenameRequirements := "Make sure its name is formatted as: DD-MM-YYY-slug.md"
 
 	if len(f) < 12 {
-		fmt.Println(errorDescription)
-		os.Exit(1)
+		return fmt.Errorf("Length of the file is too short. %s", filenameRequirements)
```
Comment (freetonik) [inline underblog.go]: Length of the file sounds like the length of its contents, not filename.

File: underblog.go [hunk 3]
```diff
@@ -67,128 +78,138 @@ func main() {
...
 	}
 
 	// day is int?
 	_, err := strconv.Atoi(f[0:2])
 	if err != nil {
-		fmt.Println(errorDescription)
-		os.Exit(1)
+		return fmt.Errorf("Day doesn't look right. %s", filenameRequirements)
```
Comment (freetonik) [inline underblog.go]: Now it reads:

```
Day doesn't look right. [..] Make sure its name is formatted as: DD-MM-YYY-slug.md
```

"Make sure its name is... == make sure day's name is...".

File: underblog.go (hunk 1 above)
Comment (velppa) [reply underblog.go:145]: I just realized that:
- typo in "YYY"
- `DD-MM-YYYY` layout doesn't preserve calendar ordering of the files, so better layout would be `YYYY-MM-DD`

**Comment by velppa** [comment]
Now it's way behind `master`, closing.

**Pull request closed without merging**